tests/test_templates_layers.py

```diff
@@ -415,15 +387,12 @@ def test_random_layer_weights(self, n_subsystems, tol):
         dev = qml.device('default.qubit', wires=n_subsystems)
         weights = np.random.randn(n_rots)
 
-        def circuit(weights):
+        with qml.utils.OperationRecorder() as rec:
             _random_layer(weights=weights, wires=range(n_subsystems), ratio_imprim=0.3,
                           imprimitive=qml.CNOT, rotations=[RX, RY, RZ], seed=4)
             return qml.expval(qml.PauliZ(0))
 
-        qnode = qml.QNode(circuit, dev)
-        qnode(weights)
-        queue = qnode.queue
-        params = [q.parameters for q in queue]
+        params = [q.parameters for q in rec]
```

Comment:
Nice, these tests have been made much simpler! Not only that, but they use fewer areas of the codebase to test the same things.

Reply:
Yes! It's way better to use the ``Recorder``!